docs/deep-dives/mobile/behavioral-targeting.mdx

```diff
@@ -0,0 +1,156 @@
+
+Based on the current datetime, the buckets may not advance at all, or may advance so much that all the buckets are cleared. As an example, if the current date for the `Hours` bucket is set to `May 1 10:00:00 UTC`, and an event is recorded at `May 1 12:45:00 UTC`, the buckets will be advanced 2 positions, the current date will be updated to `May 1 12:00:00 UTC`, and the event will be recorded in the bucket for the 12pm hour.
+
+#### Retention
+
+Each time interval has a maximum number of buckets it retains. As the time intervals move forward, buckets are rotated off of the deque and new buckets are added. If a query is performed that would go beyond the bucket count, it instead is cut off at the bucket count.
+
+The following is a list of the time intervals and their bucket counts:
+
+```
+Minutes 60
+Hours   24
+Days    56
+Weeks   52
+Months  12
+Years   4
+```
+
+## Querying for User Behavior
+
+User behaviors are recorded in the same way as Glean events, and there are a number of ways in which they can be queried.
+
+The following is a list of jexl transforms that exist within the Nimbus targeting helper, and thus are usable on all projects that use the Nimbus Rust library.
+
+| Transform | Description | Args | Returns |
+|:---|:---|:---|:---|
+| `eventSum` | Calculates the sum of all bucket values within the range | `interval, bucket_count, starting_bucket` | int |
+| `eventCountNonZero` | Calculates the total number of buckets with a non-zero value within the range | `interval, bucket_count, starting_bucket` | int |
+| `eventAverage` | Calculates the average of all event bucket values within the range | `interval, bucket_count, starting_bucket` | float |
+| `eventAveragePerNonZeroInterval` | Calculates the average of all buckets with a non-zero value within the range | `interval, bucket_count, starting_bucket` | float |
+| `eventLastSeen` | Returns the number of whole time intervals between the starting bucket and the first bucket with a non-zero value | `interval, starting_bucket` | int |
+
+## Designing Experiments
+
+The following are the existing options for behavioral targeting as defined in Experimenter.
+
+| Name | Description | Targeting String |
+|:---|:---|:---|
+| Core Active Users | A user who has used the application 21 out of the last 28 days. | <pre><code>'app-opened-event'&#124;eventCountNonZero('Days', 28, 0) >= 21</code></pre> |
+| Recently Logged In Users | A user who has logged into Sync within the last 12 weeks. | <pre><code>'sync-signin-event'&#124;eventCountNonZero('Weeks', 12, 0) > 0</code></pre> |
+
+
+## Engineering
+
+In order to instrument a new behavior/event, an equivalent call to the Nimbus event recording method must be made alongside the call to record a Glean event.
```

Comment:
Worth adding a sidebar to say that this is a temporary arrangement, until Glean allow Nimbus to mirror Glean events.